add __config__.py; *.so and *.c to .gitignore

Rahmspinat opened the pull request:
add   `__config__.py` ;  *.so  and  *.c to .gitignore

Commit:
add __config__.py; *.so and *.c to .gitignore

ax3l commented:
I would not ignore `*.c` files because files such as [`yt/utilities/data_point_utilities.c`](https://github.com/openPMD/openPMD-yt/blob/dev-1.0.0/yt/utilities/data_point_utilities.c) seem not to be created at runtime but are necessary source code. Is that correct?

What is the `__config__.py` file doing? When is it created and where? (It's not in the current source so I guess it's really temporary and fine to ignore.)

`*.so` files are compiled, shared libraries and not included in the current source, so that is fine.

ax3l commented on .gitignore:
```diff
@@ -3,3 +3,6 @@
 __pycache__
+__config__.py
+*.so
+*.c
```
that line concerns me

PrometheusPi replied:
This is definitly a problem, because in developer mode several `*.c` files are created (like `yt/geometry/grid_container.c`) and should not become part of the git repo but other `*.c` files are part of the original yt repo (like `yt/analysis_modules/halo_finding/fof/fof_main.c`).

ax3l replied:
argh, cython.

probably we have to add [cython](https://github.com/cython/cython) dirs (files with `.pyx`) manually.
update: dang, even in dirs like `yt/utilities/lib/` both are mixed (intermediate `.c` files and actual source files)

PrometheusPi committed:
fix *.c files in .gitignore

This pull request removes ignoring all *.c files
and only ignores *.c files created during a developer setup of yt.

Rahmspinat committed:
Merge pull request #2 from PrometheusPi/fix_gitignore_c_files

fix *.c files in .gitignore

PrometheusPi commented:
@ax3l [this](https://github.com/Rahmspinat/openPMD-yt/commit/0df5f8ccaf465c3859931136ba6b13692205da75) should fix the cython problem.

ax3l commented:
let us add `*.pyo` below `*.pyc`, too ([link](https://stackoverflow.com/questions/8822335/what-do-the-python-file-extensions-pyc-pyd-pyo-stand-for/8822377#8822377))

ax3l commented:
@PrometheusPi ok, should at least work until the next upstream release ;)

ax3l commented:
if you need `*.pyo` to be ignored at some point just open up a follow-up PR.

well done! :sparkles:
